pkg/kvstore/etcd.go

```diff
@@ -101,12 +138,33 @@ func NewEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 				ec.session = newSession
 				ec.sessionMU.Unlock()
 				log.Debugf("Renewing etcd session")
+				ec.CheckMinVersion()
 			}
 		}
 	}()
 	return ec, nil
 }
 
+// CheckMinVersion checks the minimal version running on etcd cluster. If the
+// minimal version running doesn't meet cilium minimal requirements, exits
+// cilium.
+func (e *EtcdClient) CheckMinVersion() {
+	eps := e.cli.Endpoints()
+	success := false
+	for _, ep := range eps {
+		if err := checkMinVersionOfEndpoint(ep); err != nil {
+			log.Debugf("Unable to check min version: %s", err)
+		} else {
+			success = true
+			break
+		}
+	}
+	if !success {
+		log.Warningf("Unable to check etcd's cluster version."+
```

Comment:
Bail out should be here only if no endpoints which meet the required spec have been found. This should not be a total bail out but instead it attempt to reconnect later. We don't want to just stop all existing workloads.